### PR TITLE
pipe: support append

### PR DESCRIPTION
**Pull request by dormanze**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
As the title.

## How to test this PR?

1. creat file
```
echo "test1" | mc pipe --append myminio/bucket/test.file
```
2. upload new content with append
```
echo "test2" | mc pipe --append myminio/bucket/test.file
```
3. cat test.file
```
mc cat myminio/bucket/test.file
```

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/pipe-main.go [hunk 1]
```diff
@@ -95,20 +100,36 @@ EXAMPLES:
   3. Copy an ISO image to an object on Amazon S3 cloud storage.
      {{.Prompt}} cat debian-8.2.iso | {{.HelpName}} s3/opensource-isos/gnuos.iso
 
-  4. Stream MySQL database dump to Amazon S3 directly.
+  4. Write contents of stdin append to an object on Amazon S3 cloud storage.
```
Comment (shtripat) [inline cmd/pipe-main.go:103]:
```suggestion
  4. Append contents of stdin to an existing object on Amazon S3 cloud storage.
```

**Comment by klauspost** [comment]
I am not against the feature per se - but please make it *very* clear in the documentation that this works by reading back the existing object. 

Users will expect to be able to append to a 1GB file instantly - and it should be noted everywhere that this will not be the case. (and even so users will still complain).

**Comment by harshavardhana** [comment]
I think this may not be what we want to encourage as a practice @dormanze it is better called as `mc compose` or something where you create a new object with existing 2 or more objects.

**Comment by dormanze** [comment]
> I think this may not be what we want to encourage as a practice @dormanze it is better called as `mc compose` or something where you create a new object with existing 2 or more objects.

That's a good idea. I'll put it on my to-do list. However, I added the append functionality initially because Linux streaming supports appending. Do you think this is necessary?

**Comment by dormanze** [comment]
> I am not against the feature per se - but please make it _very_ clear in the documentation that this works by reading back the existing object.
> 
> Users will expect to be able to append to a 1GB file instantly - and it should be noted everywhere that this will not be the case. (and even so users will still complain).

I'll add some notes to the documentation.Online documentation also needs attention.

**Pull request closed without merging**